pennylane/templates/subroutines/trotter.py

```diff
@@ -189,6 +189,8 @@ def __init__(  # pylint: disable=too-many-arguments
                 raise ValueError(
                     "There should be at least 2 terms in the Hamiltonian. Otherwise use `qml.exp`"
                 )
+            if qml.QueuingManager.recording():
```

Comment:
Thought I fixed this last week in https://github.com/PennyLaneAI/pennylane/pull/5524

But it looks like the legacy H hamiltonian conversion stuff makes it so that didn't actually fix the problem.  

We may also have this queuing issue with the `SProd` logic right below too.

Reply:
It was fixed for `Sum`, but for `Hamiltonian` and `LinearCombination`, `qml.dot` is called on here which queues another op, which does get removed by the logic added by your PR, but the original `Hamiltonian`/`LinearCombination` still remains in the queue.

Reply:
I made an update to extend this fix to `SProd` as well.